ios/FluentUI/Controls/Button.swift

```diff
@@ -212,6 +281,47 @@ open class Button: UIButton {
         }
     }
 
+    private func normalTitleAndImageColor(for window: UIWindow) -> UIColor {
+        return style == .primaryFilled ? Colors.Button.titleWithFilledBackground : Colors.primary(for: window)
+    }
+
+    private func highlightedTitleAndImageColor(for window: UIWindow) -> UIColor {
+        return style == .primaryFilled ? normalTitleAndImageColor(for: window) : Colors.primaryTint20(for: window)
+    }
+
+    private func disabledTitleAndImageColor(for window: UIWindow) -> UIColor {
+        return style == .primaryFilled ? normalTitleAndImageColor(for: window) : Colors.primaryTint20(for: window)
+    }
+
+    private var normalImageTintColor: UIColor?
+    private var highlightedImageTintColor: UIColor?
+    private var disabledImageTintColor: UIColor?
+
+    private var isUsingCustomContentEdgeInsets: Bool = false
+    private var isAdjustingCustomContentEdgeInsetsForImage: Bool = false
```

Comment:
Why do we need this bool? I might be dumb but it almost feels like a semaphore which I'm not sure why we need?

Reply:
Without this, we end up in infinite recursion.
The contentEdgeInsets can be set to a custom value. We implement the didSet property observer of contentEdgeInsets. In the didSet, we need to update the value of contentEdgeInsets if an image is already set so that we can add some padding between the image and the text. The contentEdgeInsets value will be reflected in the UI because we use titleEdgeInsets to offset the text and make it look like the original contentEdgeInsets values are respected. If we modify contentEdgeInsets in its own didSet callback, we will end up calling again into the didSet callback and it will never end.